### PR TITLE
pl$head, pl$mean, pl$median

### LINKED ISSUE #144: `pl$count()`, `pl$mean()` and other context expressions
`pl$count()` does not appear to be implemented.

I am translating all the examples in [the Python Polars User Guide](https://pola-rs.github.io/polars-book/user-guide/) to R, and a lot of their examples use this kind of construction:

```r
pl$DataFrame(mtcars)$groupby("cyl")$agg(pl$count())
```

### PR DESCRIPTION
**Pull request by vincentarelbundock**
Closes #144. Examples:

``` r
library(polars)

df = pl$DataFrame(mtcars)

df$groupby("cyl")$agg(pl$mean("mpg", "hp"))
# shape: (3, 3)
# ┌─────┬───────────┬────────────┐
# │ cyl ┆ mpg       ┆ hp         │
# │ --- ┆ ---       ┆ ---        │
# │ f64 ┆ f64       ┆ f64        │
# ╞═════╪═══════════╪════════════╡
# │ 6.0 ┆ 19.742857 ┆ 122.285714 │
# │ 4.0 ┆ 26.663636 ┆ 82.636364  │
# │ 8.0 ┆ 15.1      ┆ 209.214286 │
# └─────┴───────────┴────────────┘

df$select(pl$median("mpg"))
# shape: (1, 1)
# ┌──────┐
# │ mpg  │
# │ ---  │
# │ f64  │
# ╞══════╡
# │ 19.2 │
# └──────┘

pl$mean(df)
# shape: (1, 11)
# ┌───────────┬────────┬────────────┬──────────┬─────┬────────┬─────────┬────────┬────────┐
# │ mpg       ┆ cyl    ┆ disp       ┆ hp       ┆ ... ┆ vs     ┆ am      ┆ gear   ┆ carb   │
# │ ---       ┆ ---    ┆ ---        ┆ ---      ┆     ┆ ---    ┆ ---     ┆ ---    ┆ ---    │
# │ f64       ┆ f64    ┆ f64        ┆ f64      ┆     ┆ f64    ┆ f64     ┆ f64    ┆ f64    │
# ╞═══════════╪════════╪════════════╪══════════╪═════╪════════╪═════════╪════════╪════════╡
# │ 20.090625 ┆ 6.1875 ┆ 230.721875 ┆ 146.6875 ┆ ... ┆ 0.4375 ┆ 0.40625 ┆ 3.6875 ┆ 2.8125 │
# └───────────┴────────┴────────────┴──────────┴─────┴────────┴─────────┴────────┴────────┘

pl$mean(pl$Series(1:10))
# polars Series: shape: (1,)
# Series: '' [f64]
# [
#   5.5
# ]
```

### CODE REVIEW

File: .Rbuildignore [hunk 1]
```diff
@@ -13,3 +13,4 @@
 ^README\.Rmd$
 ^NEWS\.0\.md$
 ^docs$
+^\.venv$
```
Comment (sorhawell) [inline .Rbuildignore:16]: one extra blank line

**Review by sorhawell — APPROVED** [review]
looks very good :)

What is your opinion on the pcase, Ok, Err, unwrap  error handling?

**Comment by sorhawell** [comment]
@vincentarelbundock on a side note r-polars return Series on most Series aggregation methods, [whereas py-polars return values ](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.mean.html#polars.Series.mean)


we could do something like this, what do you think?
```R
# current
Series_mean = "use_extendr_wrapper"

# alternative
Series_mean = function() {
  .pr$Series$mean()$to_r_vector()
}
```

File: .Rbuildignore (hunk 1 above)
Comment (vincentarelbundock) [reply .Rbuildignore]: https://github.com/pola-rs/r-polars/pull/179/commits/6c889deb5aa6d101c82f4366f9f93b244d5e8466

**Comment by vincentarelbundock** [comment]
> What is your opinion on the pcase, Ok, Err, unwrap error handling?

I think it all works pretty seamlessly, and the error messages are informative, which is very nice. Now that I've looked at the actual function definitions of unwrap() et al, I have a better sense of how to use them, and it makes a lot of sense.

With respect to `pcase`, I often find that if{}else{} are more readable, but that's an aesthetic view, not an implementation issue. `pcase` itself is very easy to use (I was familiar with `data.table::fcase`.

> @vincentarelbundock on a side note r-polars return Series on most Series aggregation methods, [whereas py-polars return values ](https://pola-rs.github.io/polars/py-polars/html/reference/series/api/polars.Series.mean.html#polars.Series.mean)

I don't have a strong view on this. On the one hand, there may be computational benefits of working with Series. Say we aggregate with many groups and then want to operate on the resulting (still long) Series. And in general I like that we don't convert objects without asking (similar to the argument we had over `collect()` in S3 methods). On the other hand, consistency with python is nice, though I'm not sure it's crucial in this case.

**Comment by vincentarelbundock** [comment]
@sorhawell scratch what I said above about aggregation. Wrote while cooking and distracted; my comment doesn't make sense.

I don't have a strong view. Maybe compatibility with Python is good. If that's what you prefer, I can try to make the changes you suggested above on a large scale in the appropriate files.

**Pull request merged**